src/common/types/column_data_collection.cpp

```diff
@@ -115,19 +115,27 @@ idx_t ColumnDataRow::RowIndex() const {
 //===--------------------------------------------------------------------===//
 // ColumnDataRowCollection
 //===--------------------------------------------------------------------===//
-ColumnDataRowCollection::ColumnDataRowCollection(const ColumnDataCollection &collection) {
+ColumnDataRowCollection::ColumnDataRowCollection(const ColumnDataCollection &collection,
+                                                 ColumnDataScanState &scan_state) {
 	if (collection.Count() == 0) {
 		return;
 	}
 	// read all the chunks
-	ColumnDataScanState scan_state;
-	collection.InitializeScan(scan_state);
+	ColumnDataScanState temp_scan_state;
```

Comment:
`ColumnDataRowCollection` should create a fully materialized copy of the `ColumnDataCollection` - perhaps we can copy over the data during the scan instead, e.g. by passing in `ColumnDataScanProperties::DISALLOW_ZERO_COPY` to the `InitializeScan`? Then we can remove the scan_state argument and simplify the interface back to what it was.